llama_index/vector_stores/pinecone.py

```diff
@@ -210,21 +213,30 @@ def __init__(
     ) -> None:
         """Initialize params."""
         try:
-            import pinecone
+            # import pinecone
+            pinecone = __import__("pinecone")
         except ImportError:
             raise ImportError(import_err_msg)
 
         if pinecone_index is not None:
-            self._pinecone_index = cast(pinecone.Index, pinecone_index)
+            self._pinecone_index = cast(pinecone.Index, pinecone_index)  # type: ignore[name-defined]
```

Comment:
Might also be good to have a one or two line comment here to explain the context around this code.

Reply:
Okay, so to be totally honest, I don't quite understand why we need this line. I tried to remove it, and things failed, so we def need it.... just not sure why

Reply:
Interesting! We should dig in a bit and try to find out why...I'll investigate a bit and let you know what I find.

Reply:
Okay figured it out. This is simply for type checking. We need to cast `pinecone_index` into the type `pinecone.Index` because that's the expected type of `self._pinecone_index`. Added a comment reflecting that